### ISSUE TITLE
main.js

### ISSUE THREAD

**Issue by Wolfeitz** [Mr-Antivirus/React-Isomorphic-Bootstrap-Base#1]
Your template is almost exactly what I was hoping to find - minimalistic but has the basics and server side rendering.  Everything seems to work great on the client however when I followed the instructions and commented out main.js, nothing happened.  Nothing at all...  Click on Flux Demo and you get an infinite spinner.  Click on the other pages and you get about the same.
Is there something I'm doing wrong maybe or could this be a Windows thing?

**Comment by mrantivirus** [comment]
Hi, thanks for the compliment. I just want to point out that this project is still a work in progress. 

TLDR:
The server renders the initial state of the component before any data gets to it.

Long Story:
The reason that you see the spinning circle is because the initial state of the LocationStore is empty. The logic in the location page says to render the spinner if it doesn't have any data currently to display. So when the server gets the request for that page, it is just rendering that initial state (the spinner). Since you commented out the main.js file, there is no JavaScript to run to pull in the data, so the spinner just sits there.

I haven't gotten around to updating this project, but to solve the issue, just wait until you gather all the necessary data before you render and send to the client. That way, SEO will work like it should.

**Comment by mrantivirus** [comment]
Alright. The code change has been pushed into the master branch. I really should have made a pull request for it, shame on me. I left a quick comment in there to explain the new implementation. 

As of now, the routing waits until the store gathers its data. At which point the actions will inform the router that it is safe to proceed loading the rest of the component hierarchy.

I should note that this is only one of the few examples that I saw. I left a link in the comments inside of location.js for references.

**Comment by Wolfeitz** [comment]
Thanks for the updates!! 
I'm still in the early stages of my React learning so I really appreciate the effort put into the project.
